### PR TITLE
HADOOP-15958. Revisiting LICENSE and NOTICE files

### PR DESCRIPTION
**Pull request by aajisaka**
JIRA: https://issues.apache.org/jira/browse/HADOOP-15958

### CODE REVIEW

**Comment by aajisaka** [comment]
I'll update the dependencies shortly.

File: LICENSE-binary [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+------------
+
+hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/bloom/*
+hadoop-common-project/hadoop-common/src/main/native/gtest/gtest-all.cc
+hadoop-common-project/hadoop-common/src/main/native/gtest/include/gtest/gtest.h
+hadoop-common-project/hadoop-common/src/main/native/src/org/apache/hadoop/util/bulk_crc32_x86.c
+hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/fuse-dfs/util/tree.h
+hadoop-tools/hadoop-sls/src/main/html/js/thirdparty/d3.v3.js
+hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/static/d3-3.5.17.min.js
+leveldb
```
Comment (iwasakims) [inline LICENSE-binary]: no version specified.

File: LICENSE-binary [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/static/dust-helpers-1.1.1.min.js
+hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/static/moment.min.js
+hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/static/angular-nvd3-1.0.9.min.js
+hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/static/angular-1.6.4.min.js
+hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/static/angular-route-1.6.4.min.js
+hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/static/jquery-3.3.1.min.js
+hadoop-tools/hadoop-sls/src/main/html/js/thirdparty/jquery.js
+hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/resources/webapps/static/jquery
+hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/resources/webapps/static/jt/jquery.jstree.js
+hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/static/bootstrap-3.3.7
```
Comment (iwasakims) [inline LICENSE-binary]: These lines are not covered in LICENSE.txt. These could be sorted with above lines.

File: LICENSE-binary (hunk 1 above)
Comment (aajisaka) [reply LICENSE-binary]: I don't know the version of the leveldb libraries included in leveldbjni-all.jar.
There is no information in the GitHub repository: https://github.com/fusesource/leveldbjni
It seems that the leveldb version depends on the release manager's environment.

File: LICENSE-binary (hunk 2 above)
Comment (aajisaka) [reply LICENSE-binary]: Sorted. Thanks.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 43 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 3 | No case conflicting files found. |
| 0 | shelldocs | 3 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 70 | Maven dependency ordering for branch |
| +1 | mvninstall | 1049 | trunk passed |
| +1 | compile | 1130 | trunk passed |
| +1 | mvnsite | 844 | trunk passed |
| +1 | shadedclient | 602 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 356 | trunk passed |
| -0 | patch | 980 | Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 28 | Maven dependency ordering for patch |
| +1 | mvninstall | 1075 | the patch passed |
| +1 | compile | 1053 | the patch passed |
| +1 | javac | 1053 | the patch passed |
| +1 | mvnsite | 942 | the patch passed |
| +1 | shellcheck | 0 | There were no new shellcheck issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 7 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 665 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 370 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 8937 | root in the patch failed. |
| +1 | asflicense | 51 | The patch does not generate ASF License warnings. |
| | | 17394 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.yarn.server.nodemanager.amrmproxy.TestFederationInterceptor |
|   | hadoop.hdfs.TestStripedFileAppend |
|   | hadoop.hdfs.server.namenode.TestFsck |
|   | hadoop.hdfs.TestDecommissionWithStriped |
|   | hadoop.hdfs.TestSafeModeWithStripedFile |
|   | hadoop.hdfs.server.datanode.TestDirectoryScanner |
|   | hadoop.hdfs.server.namenode.TestNameNodeMetadataConsistency |
|   | hadoop.hdfs.TestUnsetAndChangeDirectoryEcPolicy |
|   | hadoop.hdfs.TestDistributedFileSystem |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1307/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1307 |
| Optional Tests | dupname asflicense shellcheck shelldocs compile javac javadoc mvninstall mvnsite unit shadedclient xml |
| uname | Linux f247041d42ca 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 30ce854 |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1307/5/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1307/5/testReport/ |
| Max. process+thread count | 4658 (vs. ulimit of 5500) |
| modules | C: hadoop-assemblies hadoop-tools/hadoop-sls . U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1307/5/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 51 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 2 | No case conflicting files found. |
| 0 | shelldocs | 2 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 70 | Maven dependency ordering for branch |
| +1 | mvninstall | 1114 | trunk passed |
| +1 | compile | 1231 | trunk passed |
| +1 | mvnsite | 887 | trunk passed |
| +1 | shadedclient | 628 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 415 | trunk passed |
| -0 | patch | 1064 | Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 27 | Maven dependency ordering for patch |
| +1 | mvninstall | 1116 | the patch passed |
| +1 | compile | 1090 | the patch passed |
| +1 | javac | 1090 | the patch passed |
| +1 | mvnsite | 890 | the patch passed |
| +1 | shellcheck | 1 | There were no new shellcheck issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 5 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 670 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 370 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 8409 | root in the patch failed. |
| +1 | asflicense | 48 | The patch does not generate ASF License warnings. |
| | | 17181 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.TestErasureCodingPolicies |
|   | hadoop.hdfs.TestFileChecksum |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1307/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1307 |
| Optional Tests | dupname asflicense shellcheck shelldocs compile javac javadoc mvninstall mvnsite unit shadedclient xml |
| uname | Linux d08200ea91bc 4.4.0-157-generic #185-Ubuntu SMP Tue Jul 23 09:17:01 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / ee7c261 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1307/7/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1307/7/testReport/ |
| Max. process+thread count | 4094 (vs. ulimit of 5500) |
| modules | C: hadoop-assemblies hadoop-tools/hadoop-sls . U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1307/7/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: LICENSE-binary (hunk 1 above)
Comment (iwasakims) [reply LICENSE-binary]: Hadoop bundles [leveldbjni-all-1.8.jar](https://mvnrepository.com/artifact/org.fusesource.leveldbjni/leveldbjni-all/1.8) which contains pre-built libraries for each platform. [leveldb is build from the code in github.com/chirino/leveldb](https://github.com/fusesource/leveldbjni/blob/leveldbjni-1.8/readme.md#build-procedure) and its version is [1.13](https://github.com/chirino/leveldb/blob/leveldbjni-1.8/Makefile#L73-L74).

Comment (aajisaka) [reply LICENSE-binary]: Thank you for the information! Included the version.

File: LICENSE-binary [hunk 3]
```diff
@@ -0,0 +1,532 @@
...
+javax.xml.bind:jaxb-api:2.2.11
+
+
+Eclipse Public License 1.0
+--------------------------
+
+junit:junit:4.12
+
+
+Eclipse Public License 2.0
```
Comment (iwasakims) [inline LICENSE-binary]: should be CDDL + GPLv2 with classpath exception?

**Comment by iwasakims** [comment]
I built dists with latest patch. Assembly part looks good overall.
* built tarball and source-taball by "mvn package -Psrc,dist".
* ran dev-support/bin/verify-license-files.
* tarball and source-tarball built from source-tarball contains L&N files.

**Comment by iwasakims** [comment]
CDDLs are not contained in licenses-binary.

File: LICENSE-binary (hunk 3 above)
Comment (aajisaka) [reply LICENSE-binary]: Yes. Thank you for your comment.
The libraries have been moved to eclipse repository and now the license is EPL, however, Apache Hadoop is using the old version of the libraries and the licenses were CDDL + GPLv2 with classpath exception.

**Comment by aajisaka** [comment]
> CDDLs are not contained in licenses-binary.

Added licenses-binary/LICENSE-cddl-gplv2-ce.txt

**Comment by iwasakims** [comment]
+1. Thanks for the update. Further nits (of original LICENSE.txt) should be fixed in follow-ups. Once the split is done, reviewing the update would be much easier.

**Comment by aajisaka** [comment]
Committed. Thank you, @iwasakims !

**Pull request closed without merging**